bouncer/embed_detector.py

```diff
@@ -0,0 +1,45 @@
+import fnmatch
+import re
+from urllib.parse import urlparse
+
+"""
+Hardcoded URL patterns where client is assumed to be embedded.
+
+Only the hostname and path are included in the pattern. The path must be
+specified; use "example.com/*" to match all URLs on a particular domain.
+
+Patterns are shell-style wildcards ('*' matches any number of chars, '?'
+matches a single char).
+"""
```

Comment:
Attribute docstrings are actually supposed to go _below_ the attribute, not above it. I'm not sure whether or not tools (e.g. Sphinx) will pick this up if it's above the attribute. To be honest, though, this isn't public API (not even in the sense of API that this module exports for other bouncer modules to use), it's only used within this module, so I think a simple code comment rather than a docstring would be fine.

Reply:
OK, I'll make it a comment.